pkg/kubelet/metrics/collectors/resource_metrics.go

```diff
@@ -235,13 +243,26 @@ func (rc *resourceMetricsCollector) collectContainerMemoryMetrics(ch chan<- metr
 }
 
 func (rc *resourceMetricsCollector) collectContainerSwapMetrics(ch chan<- metrics.Metric, pod summary.PodStats, s summary.ContainerStats) {
-	if s.Swap == nil || s.Swap.SwapUsageBytes == nil {
+	if s.Swap == nil {
 		return
 	}
 
-	ch <- metrics.NewLazyMetricWithTimestamp(s.Swap.Time.Time,
-		metrics.NewLazyConstMetric(containerSwapUsageDesc, metrics.GaugeValue,
-			float64(*s.Swap.SwapUsageBytes), s.Name, pod.PodRef.Name, pod.PodRef.Namespace))
+	if s.Swap.SwapUsageBytes != nil {
+		ch <- metrics.NewLazyMetricWithTimestamp(s.Swap.Time.Time,
+			metrics.NewLazyConstMetric(containerSwapUsageDesc, metrics.GaugeValue,
+				float64(*s.Swap.SwapUsageBytes), s.Name, pod.PodRef.Name, pod.PodRef.Namespace))
+	}
+
+	if s.Swap.SwapAvailableBytes != nil {
+		swapUsageBytes := float64(0)
+		if s.Swap.SwapUsageBytes != nil {
+			swapUsageBytes = float64(*s.Swap.SwapUsageBytes)
+		}
```

Comment:
small nit, but we can avoid the double `if` if we set swapUsageBytes in the stanza before this (initializing the float64(0) above line 250. thoughts?